config/v1/types_cluster_version.go

```diff
@@ -215,6 +222,58 @@ type UpdateHistory struct {
 // ClusterID is string RFC4122 uuid.
 type ClusterID string
 
+// ClusterVersionCapabilitiesDefault is a constant representing the
+// default policy for optional, core cluster components.
+type ClusterVersionCapabilitiesDefault string
+
+const (
+	// IncludeClusterVersionCapabilities indicates that optional,
+	// core cluster components should be included by default.
+	IncludeClusterVersionCapabilities = "Include"
+
+	// ExcludeClusterVersionCapabilities indicates that optional,
+	// core cluster components should be excluded by default.
+	ExcludeClusterVersionCapabilities = "Exclude"
+)
+
+// ClusterVersionCapabilitiesSpec configures the installation of
+// optional, core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesSpec struct {
+	// inclusionDefault declares the default policy for optional,
+	// core cluster components.
+
+	// +optional
+	InclusionDefault ClusterVersionCapabilitiesDefault `json:"inclusionDefault,omitempty"`
```

Comment:
instead of this, how about having it like admission.   There's 
1. a known list of "all the things"
2. a different list of "these are the ones enabled by default
3. cluster-admin sets a list of "definitely enable these"
4. cluster-admin sets a list of "definitely disable these"

That seems sufficient for all cases.

Reply:
> That seems sufficient for all cases.

it doesn't handle what happens when you upgrade your cluster and the new cluster brings a new capability.  Does that new capability get enabled, or disabled, on my cluster?  And how do i choose/override that behavior?

Reply:
> it doesn't handle what happens when you upgrade your cluster and the new cluster brings a new capability. Does that new capability get enabled, or disabled, on my cluster? And how do i choose/override that behavior?

see https://github.com/openshift/api/pull/1106#discussion_r799037431 where I finally figured out what this was attempting to do, but I think the kube API handling where we say, "latest" or "1.22" or "1.23" is a better API for that.

Reply:
As of 06f44f95 we have `None`, `v4.11`, and `vCurrent`, which seems to be what you were asking for in this thread, so marking it resolved.